video/out/vulkan/context_display.c

```diff
@@ -278,14 +285,84 @@ struct priv {
     struct vulkan_display_opts *opts;
     uint32_t width;
     uint32_t height;
+
+#if HAVE_DRM
+    struct mpv_opengl_drm_params_v2 drm_params;
+#endif
 };
 
+#if HAVE_DRM
+static bool open_render_fd(struct ra_ctx *ctx, const char *render_path)
+{
+    struct priv *p = ctx->priv;
+    p->drm_params.fd = -1;
+    p->drm_params.render_fd = -1;
+
+    int fd = open(render_path, O_RDWR | O_CLOEXEC);
+    p->drm_params.render_fd = fd;
+    return fd != -1;
+}
+
+static bool drm_setup(struct ra_ctx *ctx, int display_idx,
+                      VkPhysicalDevicePCIBusInfoPropertiesEXT *pci_props)
+{
+    const char *render_path = NULL;
+
+    drmDevice *devs[32] = {};
+    int count = drmGetDevices2(0, devs, MP_ARRAY_SIZE(devs));
+    for (int i = 0; i < count; i++) {
+        drmDevice *dev = devs[i];
+
+        MP_INFO(ctx, "Device %d: bustype: %d has_render: 0x%x\n",
+                i, dev->bustype, dev->available_nodes);
+
+        if (dev->bustype == DRM_BUS_PCI &&
+            dev->businfo.pci->domain == pci_props->pciDomain &&
+            dev->businfo.pci->bus == pci_props->pciBus &&
+            dev->businfo.pci->dev == pci_props->pciDevice &&
+            dev->businfo.pci->func == pci_props->pciFunction)
+        {
+            // Found our matching device.
+            MP_DBG(ctx, "Card is matched at %04X:%02X:%02X:%02X\n",
+                    pci_props->pciDomain, pci_props->pciBus,
+                    pci_props->pciDevice, pci_props->pciFunction);
+
+            if (!(dev->available_nodes & 1 << DRM_NODE_RENDER))
+                continue;
+
+            render_path = dev->nodes[DRM_NODE_RENDER];
+            break;
+        }
+    }
+
+    if (!render_path) {
+        MP_WARN(ctx, "Couldn't find DRM device that matches Vulkan device "
+                     "at: %04X:%02X:%02X:%02X\n",
+                     pci_props->pciDomain, pci_props->pciBus,
+                     pci_props->pciDevice, pci_props->pciFunction);
+        return false;
+    }
+
+    open_render_fd(ctx, render_path);
```

Comment:
Move the open within the loop - thus you no longer leak the drmDevices... Also "open()" can fail, depending on how the distro has it's groups setup - /dev/dri/renderD* might be root/video group only.